packages/lesswrong/server/repos/PostsRepo.ts

```diff
@@ -219,6 +219,20 @@ export default class PostsRepo extends AbstractRepo<DbPost> {
     `, [limit]);
   }
 
+  getUserDialogues(userId: string): Promise<DbPost[]> {
+    return this.any(`
+      -- PostsRepo.getUserDialogues
+      SELECT * 
+      FROM (
+          SELECT DISTINCT ON (p._id) p.* 
+          FROM "Posts" p, UNNEST("coauthorStatuses") unnested
+          WHERE p."collabEditorDialogue" IS TRUE 
+          AND ((UNNESTED->>'userId' = $1) OR (p."userId" = $1))
+      ) dialogues
+      ORDER BY "modifiedAt" DESC
+    `, [userId]);
+  }
```

Comment:
This seems both basically identical to `getMyActiveDialogues`, and also like it [doesn't need to be written out as raw SQL](https://lworg.slack.com/archives/CJUN2UAFN/p1702411602102049?thread_ts=1702407226.400639&cid=CJUN2UAFN).

It'd be something like:
```
await Posts.find({
  $or: [{ userId }, { 'coauthorStatuses.userId': userId }],
  deletedDraft: false,
  collabEditorDialogue: true
});
```

Reply:
Ah, I didn't know you could do the coauthor unnesting. Nice. (Also I rewrote it because the other query is used in a slightly different context, and was worried about overusing it for two different things in a way that woudl cause downstream headache)